<a href="https://colab.research.google.com/github/DianaRoor/-/blob/main/%D0%90%D0%BD%D0%B0%D0%BB%D0%B8%D0%B7_%D0%9C%D0%B0%D1%82%D1%80%D0%B5%D1%88%D0%BA%D0%B0.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Задание 1
В документе предоставлены реальные данные по одному из проектов для когорт,
установивших игру с 21.09.23 по 28.09.23:
https://www.matryoshka.com/tests/Ad_revenue.xlsx
Судя по этим данным, предположите, как нарастает ревеню в первый месяц жизни у
пользователей игры.
Во сколько раз увеличится ревеню нулевого дня (день установки игры), первого и
третьего дня к 30-му дню?
Пояснение к таблицe:
Лист ad revenue (данные по рекламному заработку):
Install_date - дата установки игры пользователем
Event_date - дата совершения события
Event_revenue - рекламный заработок
Platform - платформа
User_id - уникальный идентификатор пользователя
Лист installs data (данные по количеству инсталлов)
Install_date - дата
installs - количество инсталлов
Platform - платформа

In [134]:
import pandas as pd

In [135]:
#Импортирую данные
ad_revenue = pd.read_excel('Ad_revenue (1).xlsx', sheet_name='ad revenue')
installs_data = pd.read_excel('Ad_revenue (1).xlsx', sheet_name='installs data')

In [136]:
ad_revenue.head(10)

install_date event_date  event_revenue platform  \
0   2023-09-21 2023-09-21       0.316470  android   
1   2023-09-21 2023-09-21       0.063103  android   
2   2023-09-21 2023-09-21       0.316470  android   
3   2023-09-21 2023-09-21       0.054710  android   
4   2023-09-21 2023-09-21       0.024880  android   
5   2023-09-21 2023-09-21       0.065016  android   
6   2023-09-21 2023-09-21       0.000590  android   
7   2023-09-21 2023-09-21       0.374010  android   
8   2023-09-21 2023-09-21       0.035976  android   
9   2023-09-21 2023-09-21       0.068990  android   

                             user_id  
0  1695339148585-8391306266630234966  
1  1695280270975-5411644856462553821  
2  1695317488398-8939481192662600393  
3  1695317488398-8939481192662600393  
4  1695280270975-5411644856462553821  
5  1695317488398-8939481192662600393  
6  1695339967828-2936034894368767079  
7  1695317488398-8939481192662600393  
8  1695280270975-5411644856462553821  
9  1695339967828-2936034894368767079

In [137]:
#Смотрю все ли данные заполнены для корректного анализа
installs_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 579 entries, 0 to 578
Data columns (total 3 columns):
 #   Column        Non-Null Count  Dtype         
---  ------        --------------  -----         
 0   install_date  579 non-null    datetime64[ns]
 1   installs      579 non-null    int64         
 2   platform      579 non-null    object        
dtypes: datetime64[ns](1), int64(1), object(1)
memory usage: 13.7+ KB


In [138]:
installs_data.head(10)

install_date  installs platform
0   2023-09-21         1  android
1   2023-09-21         0  android
2   2023-09-21        47  android
3   2023-09-21         0  android
4   2023-09-21         0  android
5   2023-09-21         4  android
6   2023-09-21         1  android
7   2023-09-21        41  android
8   2023-09-21        11  android
9   2023-09-21         2  android

In [139]:
ad_revenue.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 78723 entries, 0 to 78722
Data columns (total 5 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   install_date   78723 non-null  datetime64[ns]
 1   event_date     78723 non-null  datetime64[ns]
 2   event_revenue  77812 non-null  float64       
 3   platform       78723 non-null  object        
 4   user_id        78723 non-null  object        
dtypes: datetime64[ns](2), float64(1), object(2)
memory usage: 3.0+ MB


In [140]:
# Объединяю данные по установкам и доходу
merged_data = pd.merge(ad_revenue, installs_data, on=['install_date','platform'], how='left')
merged_data.head(10)


install_date event_date  event_revenue platform  \
0   2023-09-21 2023-09-21        0.31647  android   
1   2023-09-21 2023-09-21        0.31647  android   
2   2023-09-21 2023-09-21        0.31647  android   
3   2023-09-21 2023-09-21        0.31647  android   
4   2023-09-21 2023-09-21        0.31647  android   
5   2023-09-21 2023-09-21        0.31647  android   
6   2023-09-21 2023-09-21        0.31647  android   
7   2023-09-21 2023-09-21        0.31647  android   
8   2023-09-21 2023-09-21        0.31647  android   
9   2023-09-21 2023-09-21        0.31647  android   

                             user_id  installs  
0  1695339148585-8391306266630234966         1  
1  1695339148585-8391306266630234966         0  
2  1695339148585-8391306266630234966        47  
3  1695339148585-8391306266630234966         0  
4  1695339148585-8391306266630234966         0  
5  1695339148585-8391306266630234966         4  
6  1695339148585-8391306266630234966         1  
7  1695339148585-8391306266630234966        41  
8  1695339148585-8391306266630234966        11  
9  1695339148585-8391306266630234966         2

In [141]:
# Добавляю столбец с днем жизни пользователя
merged_data['Day_of_life'] = (merged_data['event_date'] - merged_data['install_date']).dt.days
merged_data.head(10)

install_date event_date  event_revenue platform  \
0   2023-09-21 2023-09-21        0.31647  android   
1   2023-09-21 2023-09-21        0.31647  android   
2   2023-09-21 2023-09-21        0.31647  android   
3   2023-09-21 2023-09-21        0.31647  android   
4   2023-09-21 2023-09-21        0.31647  android   
5   2023-09-21 2023-09-21        0.31647  android   
6   2023-09-21 2023-09-21        0.31647  android   
7   2023-09-21 2023-09-21        0.31647  android   
8   2023-09-21 2023-09-21        0.31647  android   
9   2023-09-21 2023-09-21        0.31647  android   

                             user_id  installs  Day_of_life  
0  1695339148585-8391306266630234966         1            0  
1  1695339148585-8391306266630234966         0            0  
2  1695339148585-8391306266630234966        47            0  
3  1695339148585-8391306266630234966         0            0  
4  1695339148585-8391306266630234966         0            0  
5  1695339148585-8391306266630234966         4            0  
6  1695339148585-8391306266630234966         1            0  
7  1695339148585-8391306266630234966        41            0  
8  1695339148585-8391306266630234966        11            0  
9  1695339148585-8391306266630234966         2            0

In [142]:
# Фильтруюданные, нам нужны только 0, 1, 3, 30 дни
days_data = [0, 1, 3, 30]
filtered_data = merged_data[merged_data['Day_of_life'].isin(days_data)]

In [143]:
# Расчитываем ревеню на 0-й, 1-й и 3-й дни
revenue_0 = revenue_per_day.loc[0, 'event_revenue'] if 0 in revenue_per_day.index else 0
revenue_1 = revenue_per_day.loc[1, 'event_revenue'] if 1 in revenue_per_day.index else 0
revenue_3 = revenue_per_day.loc[3, 'event_revenue'] if 3 in revenue_per_day.index else 0
print(revenue_0)
print(revenue_1)
print(revenue_3)

219785.052519
90235.456025
35070.65027


In [163]:
growth_0_to_1 = abs(revenue_1 - revenue_0) / revenue_0 if revenue_0 != 0 else 0
growth_1_to_3 = abs(revenue_3 - revenue_1) / revenue_1 if revenue_1 != 0 else 0
print(growth_0_to_1)
print(growth_1_to_3)

0.5894377029247732
0.6113429042760801


In [165]:
#Расчитыва. средний рост и оценку дохода на 30й день
average_growth = abs(growth_0_to_1 + growth_1_to_3) / 2
estimated_revenue_30 = round(revenue_3 * ((1 + average_growth) ** (30 - 3)))
print(f"Средний прирост: {average_growth}")
print(f"Дохода на 30й день: {estimated_revenue_30}")

Средний прирост: 0.6003903036004266
Дохода на 30й день: 11456274919


In [166]:
multiplier_0_to_30 = estimated_revenue_30 / revenue_0 if revenue_0 != 0 else None
multiplier_1_to_30 = estimated_revenue_30 / revenue_1 if revenue_1 != 0 else None
multiplier_3_to_30 = estimated_revenue_30 / revenue_3 if revenue_3 != 0 else None

print(f"Во сколько раз увеличивается доход на 30-й день к 0-му: {round(multiplier_0_to_30, 2) if multiplier_0_to_30 is not None else None}")
print(f"Во сколько раз увеличивается доход на 30-й день к 1-му: {round(multiplier_1_to_30, 2) if multiplier_1_to_30 is not None else None}")
print(f"Во сколько раз увеличивается доход на 30-й день к 3-му: {round(multiplier_3_to_30, 2) if multiplier_3_to_30 is not None else None}")


Во сколько раз увеличивается доход на 30-й день к 0-му: 52124.9
Во сколько раз увеличивается доход на 30-й день к 1-му: 126959.79
Во сколько раз увеличивается доход на 30-й день к 3-му: 326662.75


Описание действий:
1. Импортирую файл
2. Проверяю насколько данные корректные
3. Обьединяю два листа
4. Добавляею данные идля расчета одного дня жизни пользоватебя
5. Фильтруею данные и обьединяем их с  таблицей
6. РАсчитываю ревеню 0,1 и 3 дня.
7. Так как 30 дня нет в данных, можно взять среднее значения за счет роста доходов от0 до 1 и от 1 до 3. использую уже известный доход и средний рост, оцениваю предполагаемый доход к 30му дню.
8. Расчитываю множетели, для оценки во сколько раз предполагаемый доходт на 30й день первышает доходы на 0й, 1 й и 3 й день
0. Вывожу результаты с округлением.

Таким образом:
Доход на 30-й день по отношению к 0-му увеличится в : 52124.9
Доход на 30-й день по отношению к 1-му увеличится в : 126959.79
Доход на 30-й день по отношению к 3-му увеличится в : 326662.75